CefSharp/ModelBinding/DefaultBinder.cs

```diff
@@ -144,10 +148,8 @@ protected virtual void BindValue(BindingMemberInfo modelProperty, object obj, Bi
                 return;
             }
 
-            Type dictionaryType = typeof(Dictionary<string, object>);
-
             //If the type is a dictionary and the PropertyType isn't then we'll bind.
-            if (obj.GetType() == dictionaryType && modelProperty.PropertyType != dictionaryType)
+            if (obj.GetType().IsGenericObject() && !modelProperty.PropertyType.IsGenericObject())
```

Comment:
As the two `if` branches here are duplicated I think we can simplify the whole `BindValue` method. We should be able to simplify things quite a bit, using something like the following.

```c#
protected virtual void BindValue(BindingMemberInfo modelProperty, object obj, BindingContext context)
{
	if(obj == null)
	{
		return;
	}

	if (modelProperty.PropertyType.IsAssignableFrom(obj.GetType()))
	{
		//Simply set the property
		modelProperty.SetValue(context.Model, obj);
	}
	else
	{
		//Cannot directly set the property attempt to bind
		var model = Bind(obj, modelProperty.PropertyType);

		modelProperty.SetValue(context.Model, model);
	}
}
```